### ISSUE TITLE
Menu items with checkboxes

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#1209]
There are a couple of refinements we can make to menu items that include checkboxes:

1. For disabled checkboxes, remove `aria-disabled` since `disabled` is supported natively
2. Make the area that displays a background color clickable. In general, anywhere that hover styles display should also be clickable when possible. This can be done by changing `div.pf-c-check pf-c-dropdown__menu-item` to `label.pf-c-check pf-c-dropdown__menu-item`, and `label.pf-c-check__label` to `span.pf-c-check__label`. The same class structure can still be used, just swap out the elements.

**Comment by jgiardino** [comment]
@seanforyou23 We had talked in the past about implicit labels. Looking back at the recommendations from Deque, they have this note:

> There have been some edge cases in which some browsers and/or screen readers lacked support for implicit labels in very specific circumstances, such as the `<input type="number">` element in Firefox with the NVDA screen reader opens in a new window, but overall support is solid for this method.

I think we settled with being on the conservative side, and keeping the explicit association even in this case. Just wanted to confirm if that's still what we want here.

**Comment by mcoker** [comment]
@jgiardino re: #1, I think you can strike that out. I searched the codebase and looks like we were only using `aria-disabled` with `disabled` on the `pf-c-check` component and in the multi-select example in the `pf-c-dropdown` component. I've addressed both in https://github.com/patternfly/patternfly-next/pull/1211 (check) and https://github.com/patternfly/patternfly-next/pull/1075 (dropdown)

**Comment by mcoker** [comment]
#1211 and #1075 were merged
